numpy/lib/type_check.py

```diff
@@ -134,13 +135,17 @@ def real(val):
     >>> a.real = np.array([9, 8, 7])
     >>> a
     array([ 9.+2.j,  8.+4.j,  7.+6.j])
+    >>> np.real(1 + 1j)
+    1.0
 
     """
-    return asanyarray(val).real
+    arr = asanyarray(val)
+    return arr.real.item() if _nx.isscalar(val) else arr.real
```

Comment:
Although now this is inconsistent with `np.conj`, which does not return a native `complex`...

What was the previous implementation?